packages/ra-core/src/dataProvider/useQuery.ts

```diff
@@ -1,7 +1,7 @@
 import { useEffect } from 'react';
 
 import { useSafeSetState } from '../util/hooks';
-import useDataProvider from './useDataProvider';
+import useDataProvider from './useDataProviderWithDeclarativeSideEffects';
```

Comment:
I don't want that `useQuery` accept declarative side effects. It's a new hook, we can enforce its syntax. I just want it for `<Query>`.